samples/kotlinDslExample/build.gradle.kts

```diff
@@ -0,0 +1,124 @@
+import com.google.protobuf.gradle.*
+import groovy.lang.Closure
+import groovy.lang.GroovyObject
+import org.gradle.api.internal.tasks.DefaultSourceSet
+import org.gradle.api.internal.tasks.DefaultSourceSetContainer
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+group = "gkt"
+version = "1.0-SNAPSHOT"
+
+buildscript {
+    var kotlin_version: String by extra
+    kotlin_version = "1.2.10"
+    var grpc_version: String by extra
+    grpc_version = "1.8.0"
+
+    repositories {
+        jcenter()
+    }
+
+    dependencies {
+        classpath(kotlin("gradle-plugin", kotlin_version))
+    }
+
+}
+
+val kotlin_version: String by extra
+var grpc_version: String by extra
+
+plugins {
+    application
+    java
+    idea
+    kotlin("jvm") version ("1.2.10")
+    id("com.google.protobuf") version ("0.8.3")
+}
+
+repositories {
+    jcenter()
+}
+
+dependencies {
+    compile(kotlin("stdlib-jdk8", kotlin_version))
+    compile("io.grpc:grpc-netty:$grpc_version")
+    compile("io.grpc:grpc-protobuf:$grpc_version")
+    compile("io.grpc:grpc-stub:$grpc_version")
+
+    testCompile("io.grpc:grpc-testing:$grpc_version")
+    testCompile("org.mockito:mockito-core:1.9.5")
+    testCompile("junit", "junit", "4.12")
+}
+
+configure<ProtobufConvention> {
+    protobuf(closureOf<ProtobufConfigurator> {
+        generatedFilesBaseDir = "$projectDir/src/generated"
+        protoc(closureOf<ExecutableLocator> {
+            artifact = "com.google.protobuf:protoc:3.5.0"
+        })
+        plugins(closureOf<NamedDomainObjectContainer<ExecutableLocator>> {
+            create("grpc", closureOf<ExecutableLocator> {
+                artifact = "io.grpc:protoc-gen-grpc-java:$grpc_version"
+            })
+        })
+        generateProtoTasks(closureOf<ProtobufConfigurator.JavaGenerateProtoTaskCollection> {
+            all().forEach {
+                it?.plugins(closureOf<NamedDomainObjectContainer<GenerateProtoTask.PluginOptions>> {
+                    add(create("grpc"))
+                })
+            }
+        })
+    })
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    sourceSets {
+        getByName("main") {
+            withGroovyBuilder {
+                "proto" {
+                    "srcDir"("proto")
+                }
+            }
+            java {
```

Comment:
With https://github.com/google/protobuf-gradle-plugin/commit/65b58a2fa43ab7d5dfa4fd47447d9b000b187054 I think this block is no longer necessary. I'm having some trouble getting this project to build with 0.8.4-SNAPSHOT (the version that this CL is based on).

I did a `./gradlew install` at the root directory of this project, but could not figure out the kotlin DSL equivalent of:

```
buildscript {
  repositories {
   mavenLocal()
    maven { url "https://plugins.gradle.org/m2/" }
  }
  dependencies {
    classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.4-SNAPSHOT'
  }
}
```
Can you see if the SNAPSHOT version still requires specifying the generated source dirs?

Reply:
Ah I figured out how to get the SNAPSHOT to work. It requires adding this `settings.gradle.kts` and deleting `settings.gradle` so that the `kts` version is used:

```
pluginManagement {
    repositories {
        mavenLocal()
        gradlePluginPortal()
    }
    resolutionStrategy {
        eachPlugin {
            if (requested.id.id == "com.google.protobuf") {
                useModule("com.google.protobuf:protobuf-gradle-plugin:${requested.version}")
            }
        }
    }
}
```

This is documented here: https://github.com/gradle/kotlin-dsl/blob/master/doc/getting-started/Configuring-Plugins.md